### PR TITLE
36 change gsdmvn into gsdesign2 in copyright

### PR DESCRIPTION
**Pull request by LittleBeannie**

### CODE REVIEW

File: R/check_arg.R [hunk 1]
```diff
@@ -1,3 +1,20 @@
+#  Copyright (c) 2021 Merck Sharp & Dohme Corp. a subsidiary of Merck & Co., Inc., Kenilworth, NJ, USA.
```
Comment (nanxstats) [inline R/check_arg.R:1]: 2022

Comment (nanxstats) [reply R/check_arg.R:1]: Also, the copyright line is outdated - you should use the "Rahway" version.

File: R/utility_wlr.R [hunk 2]
```diff
@@ -0,0 +1,253 @@
+#' Create "npsurvSS" arm object
```
Comment (nanxstats) [inline R/utility_wlr.R:1]: This file will need a copyright header like the others.

File: tests/testthat/test-developer-gs_spending_bound.R [hunk 3]
```diff
@@ -91,10 +90,17 @@ test_that("Two-sided asymmetric design fails to reproduce gsDesign test.type=3 b
...
   expect_equal(gsd$lower$bound, (gsdv %>% filter(Bound == "Lower"))$Z, tolerance = 9e-6)
-  expect_equal(gsd$n.I, (gsdv %>% filter(Bound == "Upper"))$info, tolerance = .04) # While tolerance should not be problematic, it seems large
+  
+  # compare statistical information
+  # While tolerance should not be problematic, it seems large
+  expect_equal(gsd$n.I, (gsdv %>% filter(Bound == "Upper"))$info, tolerance = .04) 
+  
+  # compare crossing boundaries probability under null hypothesis (theta = 0)
+  expect_equal((gsdv %>% filter(Bound == "Upper"))$Probability0, sfu(alpha = alpha, t = timing, param = sfupar)$spend)
+  #expect_equal((gsdv %>% filter(Bound == "Lower"))$Probability, sfu(alpha = beta, t = timing, param = sflpar)$spend)
```
Comment (nanxstats) [inline tests/testthat/test-developer-gs_spending_bound.R:103]: What's happening with the lines that are commented out? Please either 1) remove them if they are not useful anymore or 2) document the specific reason why they are commented out.

File: tests/testthat/test-developer-gs_spending_bound.R [hunk 4]
```diff
@@ -112,16 +118,25 @@ test_that("Two-sided asymmetric design fails to reproduce gsDesign test.type=3 b
...
   expect_equal(gsd$upper$bound, (gsdv %>% filter(Bound == "Upper"))$Z, tolerance = 7e-6)
   expect_equal(gsd$lower$bound, (gsdv %>% filter(Bound == "Lower"))$Z, tolerance = 9e-6)
-  expect_equal(gsd$n.I, (gsdv %>% filter(Bound == "Upper"))$info, tolerance = .04) # While tolerance should not be problematic, it seems large
+  
+  # compare statistical information
+  # While tolerance should not be problematic, it seems large
+  expect_equal(gsd$n.I, (gsdv %>% filter(Bound == "Upper"))$info, tolerance = .04) 
+  
+  # compare crossing boundaries probability under null hypothesis (theta = 0)
+  #expect_equal((gsdv %>% filter(Bound == "Upper"))$Probability0, sfu(alpha = alpha, t = timing, param = sfupar)$spend)
```
Comment (nanxstats) [inline tests/testthat/test-developer-gs_spending_bound.R:138]: Same.

**Review by nanxstats — CHANGES_REQUESTED** [review]
Overall looking good - needs a few minor updates such as using the latest copyright line in all files and taking care of the commented-out lines of code.

**Comment by nanxstats** [comment]
Here is the full copyright line:

`Copyright (c) 2022 Merck & Co., Inc., Rahway, NJ, USA and its affiliates. All rights reserved.`

File: tests/testthat/test-developer-gs_spending_bound.R (hunk 3 above)
Comment (LittleBeannie) [reply tests/testthat/test-developer-gs_spending_bound.R:103]: It is a pending test. I will revise it later in the test branch.

File: tests/testthat/test-developer-gs_spending_bound.R (hunk 4 above)
Comment (LittleBeannie) [reply tests/testthat/test-developer-gs_spending_bound.R:138]: It is a pending test. I will revise it later in the test branch.

**Review by nanxstats — APPROVED** [review]
OK, since this is touching so many files, let's merge it before the change piles up further.

**Pull request merged**